### ISSUE TITLE
(URGENT) Do not generate Change YAML files for "long form"

### ISSUE THREAD

**Issue by ronaldtse** [metanorma/stepmod-utils#210]
The files "arm_lf.exp" and "mim_lf.exp" are "long-form" EXPRESS schemas, and they are not tracked for changes.

We should not generate EXPRESS Change YAML files for them.

These are currently generated and also copied during `stepmod-srl-migration`. We do not want to generate them in the first place, so there is no need to copy.

e.g.
```
[stepmod-utils] Copied arm_lf.changes.yaml to schemas/work_structure/arm_lf.changes.yaml
```

**Comment by HassanAkbar** [comment]
@ronaldtse From what I understand from the conversation we do need the `*_lf.changes.yaml` files. 
Did I get this right or do we need to remove the files?

**Comment by HassanAkbar** [comment]
@TRThurman Do you have any info regarding this ⬆️ ?

**Comment by TRThurman** [comment]
Yes;Long form change logs are required.Sent from my iPhoneOn Dec 22, 2023, at 12:11 AM, HassanAkbar ***@***.***> wrote:﻿
@TRThurman Do you have any info regarding this ⬆️ ?

**Comment by HassanAkbar** [comment]
@ronaldtse As this is needed, do you think I should close this ticket, or is something else needed?

**Comment by ronaldtse** [comment]
@TRThurman thanks.

I have verified that the following files (in [iso-10303-smrl](https://github.com/metanorma/iso-10303-smrl)) contain longform changes:

ARM Longform changes:
```
reference-docs/ISO_10303-209/data/modules/ap209_multidisciplinary_analysis_and_design/sys/g_change.htm:      <h2><a name="arm_longform.changes2">F.2.4 Changes made to the ARM Longform</a></h2>
reference-docs/ISO_10303-209/data/modules/ap242_managed_model_based_3d_engineering/sys/g_change.htm:      <h2><a name="arm_longform.changes2">F.2.5 Changes made to the ARM Longform</a></h2>
reference-docs/ISO_10303-210/data/modules/ap209_multidisciplinary_analysis_and_design/sys/g_change.htm:      <h2><a name="arm_longform.changes2">F.2.4 Changes made to the ARM Longform</a></h2>
reference-docs/ISO_10303-210/data/modules/ap210_electronic_assembly_interconnect_and_packaging_design/sys/g_change.htm:      <h2><a name="arm_longform.changes4">F.4.4 Changes made to the ARM Longform</a></h2>
reference-docs/ISO_10303-210/data/modules/ap210_electronic_assembly_interconnect_and_packaging_design/sys/g_change.htm:      <h2><a name="arm_longform.changes5">F.5.4 Changes made to the ARM Longform</a></h2>
reference-docs/ISO_10303-210/data/modules/ap242_managed_model_based_3d_engineering/sys/g_change.htm:      <h2><a name="arm_longform.changes2">F.2.5 Changes made to the ARM Longform</a></h2>
reference-docs/ISO_10303-210/data/modules/ap242_managed_model_based_3d_engineering/sys/g_change.htm:      <h2><a name="arm_longform.changes3">F.3.4 Changes made to the ARM Longform</a></h2>
reference-docs/ISO_10303-210/data/modules/elemental_topology/sys/g_change.htm:      <h2><a name="arm_longform.changes6">F.6.2 Changes made to the ARM Longform</a></h2>
reference-docs/ISO_10303-242/C066654e-modified-html/C066654e-html/data/modules/ap209_multidisciplinary_analysis_and_design/sys/g_change.htm:      <h2><a name="arm_longform.changes2">F.2.4 Changes made to the ARM Longform</a></h2>
reference-docs/ISO_10303-242/C066654e-modified-html/C066654e-html/data/modules/ap210_electronic_assembly_interconnect_and_packaging_design/sys/g_change.htm:      <h2><a name="arm_longform.changes4">F.4.4 Changes made to the ARM Longform</a></h2>
reference-docs/ISO_10303-242/C066654e-modified-html/C066654e-html/data/modules/ap210_electronic_assembly_interconnect_and_packaging_design/sys/g_change.htm:      <h2><a name="arm_longform.changes5">F.5.4 Changes made to the ARM Longform</a></h2>
reference-docs/ISO_10303-242/C066654e-modified-html/C066654e-html/data/modules/ap242_managed_model_based_3d_engineering/sys/g_change.htm:      <h2><a name="arm_longform.changes2">F.2.5 Changes made to the ARM Longform</a></h2>
reference-docs/ISO_10303-242/C066654e-modified-html/C066654e-html/data/modules/ap242_managed_model_based_3d_engineering/sys/g_change.htm:      <h2><a name="arm_longform.changes3">F.3.4 Changes made to the ARM Longform</a></h2>
reference-docs/ISO_10303-242/C066654e-modified-html/C066654e-html/data/modules/elemental_topology/sys/g_change.htm:      <h2><a name="arm_longform.changes6">F.6.2 Changes made to the ARM Longform</a></h2>
reference-docs/SMRL_v8_rc5_N10480/data/modules/ap209_multidisciplinary_analysis_and_design/sys/g_change.htm:      <h2><a name="arm_longform.changes2">F.2.4 Changes made to the ARM Longform</a></h2>
reference-docs/SMRL_v8_rc5_N10480/data/modules/ap210_electronic_assembly_interconnect_and_packaging_design/sys/g_change.htm:      <h2><a name="arm_longform.changes4">F.4.4 Changes made to the ARM Longform</a></h2>
reference-docs/SMRL_v8_rc5_N10480/data/modules/ap210_electronic_assembly_interconnect_and_packaging_design/sys/g_change.htm:      <h2><a name="arm_longform.changes5">F.5.4 Changes made to the ARM Longform</a></h2>
reference-docs/SMRL_v8_rc5_N10480/data/modules/ap242_managed_model_based_3d_engineering/sys/g_change.htm:      <h2><a name="arm_longform.changes2">F.2.5 Changes made to the ARM Longform</a></h2>
reference-docs/SMRL_v8_rc5_N10480/data/modules/ap242_managed_model_based_3d_engineering/sys/g_change.htm:      <h2><a name="arm_longform.changes3">F.3.4 Changes made to the ARM Longform</a></h2>
reference-docs/SMRL_v8_rc5_N10480/data/modules/elemental_topology/sys/g_change.htm:      <h2><a name="arm_longform.changes6">F.6.2 Changes made to the ARM Longform</a></h2>
reference-docs/SMRL_v9_N11017_20221206/data/modules/ap209_multidisciplinary_analysis_and_design/sys/g_change.htm:      <h2><a name="arm_longform.changes2">F.2.4 Changes made to the ARM Longform</a></h2>
reference-docs/SMRL_v9_N11017_20221206/data/modules/ap210_electronic_assembly_interconnect_and_packaging_design/sys/g_change.htm:      <h2><a name="arm_longform.changes4">F.4.4 Changes made to the ARM Longform</a></h2>
reference-docs/SMRL_v9_N11017_20221206/data/modules/ap210_electronic_assembly_interconnect_and_packaging_design/sys/g_change.htm:      <h2><a name="arm_longform.changes5">F.5.4 Changes made to the ARM Longform</a></h2>
reference-docs/SMRL_v9_N11017_20221206/data/modules/ap210_electronic_assembly_interconnect_and_packaging_design/sys/g_change.htm:      <h2><a name="arm_longform.changes6">F.6.3 Changes made to the ARM Longform</a></h2>
reference-docs/SMRL_v9_N11017_20221206/data/modules/ap242_managed_model_based_3d_engineering/sys/g_change.htm:      <h2><a name="arm_longform.changes2">F.2.5 Changes made to the ARM Longform</a></h2>
reference-docs/SMRL_v9_N11017_20221206/data/modules/ap242_managed_model_based_3d_engineering/sys/g_change.htm:      <h2><a name="arm_longform.changes3">F.3.4 Changes made to the ARM Longform</a></h2>
reference-docs/SMRL_v9_N11017_20221206/data/modules/ap242_managed_model_based_3d_engineering/sys/g_change.htm:      <h2><a name="arm_longform.changes4">F.4.4 Changes made to the ARM Longform</a></h2>
reference-docs/SMRL_v9_N11017_20221206/data/modules/elemental_topology/sys/g_change.htm:      <h2><a name="arm_longform.changes6">F.6.2 Changes made to the ARM Longform</a></h2>
```

MIM Longform changes:
```
reference-docs/ISO_10303-209/data/modules/ap209_multidisciplinary_analysis_and_design/sys/g_change.htm:      <h2><a name="mim_longform.changes2">F.2.5 Changes made to the MIM Longform</a></h2>
reference-docs/ISO_10303-209/data/modules/ap242_managed_model_based_3d_engineering/sys/g_change.htm:      <h2><a name="mim_longform.changes2">F.2.6 Changes made to the MIM Longform</a></h2>
reference-docs/ISO_10303-210/data/modules/ap209_multidisciplinary_analysis_and_design/sys/g_change.htm:      <h2><a name="mim_longform.changes2">F.2.5 Changes made to the MIM Longform</a></h2>
reference-docs/ISO_10303-210/data/modules/ap210_electronic_assembly_interconnect_and_packaging_design/sys/g_change.htm:      <h2><a name="mim_longform.changes4">F.4.5 Changes made to the MIM Longform</a></h2>
reference-docs/ISO_10303-210/data/modules/ap210_electronic_assembly_interconnect_and_packaging_design/sys/g_change.htm:      <h2><a name="mim_longform.changes5">F.5.5 Changes made to the MIM Longform</a></h2>
reference-docs/ISO_10303-210/data/modules/ap242_managed_model_based_3d_engineering/sys/g_change.htm:      <h2><a name="mim_longform.changes2">F.2.6 Changes made to the MIM Longform</a></h2>
reference-docs/ISO_10303-210/data/modules/ap242_managed_model_based_3d_engineering/sys/g_change.htm:      <h2><a name="mim_longform.changes3">F.3.5 Changes made to the MIM Longform</a></h2>
reference-docs/ISO_10303-242/C066654e-modified-html/C066654e-html/data/modules/ap209_multidisciplinary_analysis_and_design/sys/g_change.htm:      <h2><a name="mim_longform.changes2">F.2.5 Changes made to the MIM Longform</a></h2>
reference-docs/ISO_10303-242/C066654e-modified-html/C066654e-html/data/modules/ap210_electronic_assembly_interconnect_and_packaging_design/sys/g_change.htm:      <h2><a name="mim_longform.changes4">F.4.5 Changes made to the MIM Longform</a></h2>
reference-docs/ISO_10303-242/C066654e-modified-html/C066654e-html/data/modules/ap210_electronic_assembly_interconnect_and_packaging_design/sys/g_change.htm:      <h2><a name="mim_longform.changes5">F.5.5 Changes made to the MIM Longform</a></h2>
reference-docs/ISO_10303-242/C066654e-modified-html/C066654e-html/data/modules/ap242_managed_model_based_3d_engineering/sys/g_change.htm:      <h2><a name="mim_longform.changes2">F.2.6 Changes made to the MIM Longform</a></h2>
reference-docs/ISO_10303-242/C066654e-modified-html/C066654e-html/data/modules/ap242_managed_model_based_3d_engineering/sys/g_change.htm:      <h2><a name="mim_longform.changes3">F.3.5 Changes made to the MIM Longform</a></h2>
reference-docs/SMRL_v8_rc5_N10480/data/modules/ap209_multidisciplinary_analysis_and_design/sys/g_change.htm:      <h2><a name="mim_longform.changes2">F.2.5 Changes made to the MIM Longform</a></h2>
reference-docs/SMRL_v8_rc5_N10480/data/modules/ap210_electronic_assembly_interconnect_and_packaging_design/sys/g_change.htm:      <h2><a name="mim_longform.changes4">F.4.5 Changes made to the MIM Longform</a></h2>
reference-docs/SMRL_v8_rc5_N10480/data/modules/ap210_electronic_assembly_interconnect_and_packaging_design/sys/g_change.htm:      <h2><a name="mim_longform.changes5">F.5.5 Changes made to the MIM Longform</a></h2>
reference-docs/SMRL_v8_rc5_N10480/data/modules/ap242_managed_model_based_3d_engineering/sys/g_change.htm:      <h2><a name="mim_longform.changes2">F.2.6 Changes made to the MIM Longform</a></h2>
reference-docs/SMRL_v8_rc5_N10480/data/modules/ap242_managed_model_based_3d_engineering/sys/g_change.htm:      <h2><a name="mim_longform.changes3">F.3.5 Changes made to the MIM Longform</a></h2>
reference-docs/SMRL_v9_N11017_20221206/data/modules/ap209_multidisciplinary_analysis_and_design/sys/g_change.htm:      <h2><a name="mim_longform.changes2">F.2.5 Changes made to the MIM Longform</a></h2>
reference-docs/SMRL_v9_N11017_20221206/data/modules/ap210_electronic_assembly_interconnect_and_packaging_design/sys/g_change.htm:      <h2><a name="mim_longform.changes4">F.4.5 Changes made to the MIM Longform</a></h2>
reference-docs/SMRL_v9_N11017_20221206/data/modules/ap210_electronic_assembly_interconnect_and_packaging_design/sys/g_change.htm:      <h2><a name="mim_longform.changes5">F.5.5 Changes made to the MIM Longform</a></h2>
reference-docs/SMRL_v9_N11017_20221206/data/modules/ap210_electronic_assembly_interconnect_and_packaging_design/sys/g_change.htm:      <h2><a name="mim_longform.changes6">F.6.4 Changes made to the MIM Longform</a></h2>
reference-docs/SMRL_v9_N11017_20221206/data/modules/ap242_managed_model_based_3d_engineering/sys/g_change.htm:      <h2><a name="mim_longform.changes2">F.2.6 Changes made to the MIM Longform</a></h2>
reference-docs/SMRL_v9_N11017_20221206/data/modules/ap242_managed_model_based_3d_engineering/sys/g_change.htm:      <h2><a name="mim_longform.changes3">F.3.5 Changes made to the MIM Longform</a></h2>
reference-docs/SMRL_v9_N11017_20221206/data/modules/ap242_managed_model_based_3d_engineering/sys/g_change.htm:      <h2><a name="mim_longform.changes4">F.4.5 Changes made to the MIM Longform</a></h2>
```

However, the current Changes YAML generation process completely fails to extract these changes:

https://github.com/metanorma/iso-10303-srl/blob/main/schemas/modules/ap209_multidisciplinary_analysis_and_design/arm_lf.changes.yaml

```yaml
---
schema: ap209_multidisciplinary_analysis_and_design
change_edition:
- version: '2'
  description: |-
    * Add SELECT value 'ENTITY Envelope'
    * Add SELECT value 'ENTITY Evidence'
    * Add SELECT value 'ENTITY Interface_connection'
    * Add SELECT value 'ENTITY Interface_connector_occurrence'
    * Add SELECT value 'ENTITY Interface_definition_connection'
    * Add SELECT value 'ENTITY Message'
    * Add SELECT value 'ENTITY Product_group'
    * Add SELECT value 'ENTITY Product_group_membership'
    * Add SELECT value 'ENTITY State'
    * Add SELECT value 'ENTITY State_assertion'
    * Add SELECT value 'ENTITY State_assessment'
    * Add SELECT value 'ENTITY State_relationship'
    * Add SELECT value 'ENTITY Validation'
    * Add SELECT value 'ENTITY Verification'
```

So I'll file this as a new ticket.

**Comment by HassanAkbar** [comment]
@ronaldtse The changes for the `longform` are updated. Should we close this now or is there something else that needs to be done in this ticket?

**Comment by ronaldtse** [comment]
This is fixed in #238 .
